modules/navier_stokes/src/dgkernels/MassFluxPenalty.C

```diff
@@ -0,0 +1,71 @@
+//* This file is part of the MOOSE framework
+//* https://mooseframework.inl.gov
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "MassFluxPenalty.h"
+
+// MOOSE includes
+#include "MooseVariableFE.h"
+
+#include "libmesh/utility.h"
+
+registerMooseObject("NavierStokesApp", MassFluxPenalty);
+
+InputParameters
+MassFluxPenalty::validParams()
+{
+  InputParameters params = ADDGKernel::validParams();
+  params.addRequiredCoupledVar("u", "The x-velocity");
+  params.addRequiredCoupledVar("v", "The y-velocity");
+  params.addRequiredParam<unsigned short>("component",
+                                          "The velocity component this object is being applied to");
+  params.addParam<Real>("gamma", 1, "The penalty to multiply the jump");
+  return params;
```

Comment:
missing class description. 
maybe we should delete ADDGKernel's so we get warned of these things